rootmap/sqlrootmap.go

```diff
@@ -222,35 +456,47 @@ func (rm *sqlRootmap) Get(
 	producerID string,
 	topic string,
 	version uint64,
-) (string, nodestore.NodeID, error) {
-	var nodeID string
-	var prefix string
-	err := rm.db.QueryRowContext(ctx, `
-	select storage_prefix,
-	node_id
-	from rootmap
-	where database = ? and producer_id = ? and topic = ? and version = ?`,
-		database, producerID, topic, version,
-	).Scan(&prefix, &nodeID)
+) (prefix string, nodeID nodestore.NodeID, err error) {
+	var nodeIDStr string
+	var truncationVersion *int64
+	err = rm.db.QueryRowContext(ctx, `
+	select storage_prefix, node_id, latest_truncations.version
+	from tables inner join rootmap
+	on tables.id = rootmap.table_id
+	left join latest_truncations
+	on latest_truncations.table_id = tables.id
+	where tables.database = ?
+	and tables.producer = ?
+	and tables.topic = ?
+	and rootmap.version = ?
+	`, database, producerID, topic, version,
+	).Scan(&prefix, &nodeIDStr, &truncationVersion)
 	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
-			return prefix, nodestore.NodeID{}, NewTableNotFoundError(database, producerID, topic)
+			return prefix, nodeID, NewTableNotFoundError(database, producerID, topic)
 		}
-		return prefix, nodestore.NodeID{}, fmt.Errorf("failed to read from rootmap: %w", err)
+		return prefix, nodeID, fmt.Errorf("failed to read from rootmap: %w", err)
 	}
-	decoded, err := hex.DecodeString(nodeID)
+	// if truncationVersion != nil && *truncationVersion >= int64(version) {
+	// 	return prefix, nodeID, NewRequestedRootTruncatedError(version)
+	// }
```

Comment:
I guess this is removed because the caller now supplies these to iterators? We can delete the comment